src/main/kotlin/me/fornever/todosaurus/issueTrackers/ui/wizard/ChooseIssueTrackerStep.kt

```diff
@@ -0,0 +1,319 @@
+import me.fornever.todosaurus.issueTrackers.ui.controls.IssueTrackerComboBox
+import me.fornever.todosaurus.issueTrackers.ui.controls.IssueTrackerCredentialsComboBox
+import me.fornever.todosaurus.issueTrackers.ui.controls.ServerHostComboBox
+import me.fornever.todosaurus.ui.wizard.DynamicStepProvider
+import me.fornever.todosaurus.ui.wizard.MemorableStep
+import me.fornever.todosaurus.ui.wizard.TodosaurusContext
+import me.fornever.todosaurus.ui.wizard.TodosaurusStep
+import javax.swing.JButton
+import javax.swing.JComponent
+import javax.swing.JLabel
+
+class ChooseIssueTrackerStep(private val project: Project, private val model: TodosaurusContext)
+    : TodosaurusStep(), DynamicStepProvider, MemorableStep {
+    companion object {
+        val id: Any = ChooseIssueTrackerStep::class.java
+    }
+
+    override val id: Any = Companion.id
+
+    private var issueTrackerPicker: IssueTrackerComboBox = IssueTrackerComboBox()
+    private var serverHostPicker: ServerHostComboBox = ServerHostComboBox()
+    private var credentialsPicker: IssueTrackerCredentialsComboBox = IssueTrackerCredentialsComboBox()
+    private var connectAnonymouslyCheckBox: JBCheckBox = JBCheckBox(TodosaurusBundle.message("wizard.steps.chooseIssueTracker.account.connectAnonymously"))
+    private var testConnectionResultLabel: JLabel = JLabel()
+    private lateinit var testConnectionButton: JButton
+
+    private var cachedServerPaths: Map<String, ServerPath>? = null
+    private var cachedCredentials: Map<String, List<IssueTrackerCredentials>?>? = null
+
+    init {
+        issueTrackerPicker.removeAllItems()
+
+        credentialsPicker.addItemListener {
+            testConnectionResultLabel.text = ""
+        }
+
+        serverHostPicker.addItemListener { event ->
+            val serverHost = event.item as? String ?: return@addItemListener
+
+            credentialsPicker.removeAllItems()
+            testConnectionResultLabel.text = ""
+
+            cachedCredentials
+                ?.getValue(serverHost)
+                ?.forEach {
+                    credentialsPicker.addItem(it)
+                }
+        }
+
+        issueTrackerPicker.addItemListener { event ->
+            val issueTracker = event.item as? IssueTracker ?: return@addItemListener
+
+            serverHostPicker.removeAllItems()
+            testConnectionResultLabel.text = ""
+
+            val credentials = runBlocking {
+                IssueTrackerCredentialsProviderFactory
+                    .getInstance(project)
+                    .create(issueTracker)
+                    .provideAll()
+            }
+
+            cachedServerPaths = credentials.associate {
+                it.serverPath.toURI().host to it.serverPath
+            }
+
+            cachedCredentials = credentials.groupBy {
+                it.serverPath.toURI().host
+            }
+
+            cachedServerPaths?.forEach {
+                serverHostPicker.addItem(it.key)
+            }
```

Comment:
I have found why this doesn't work for me; looks like a flaw in the logic.

You see, here we'll only show the servers we already have the stored credentials for — making it impossible for the user to enter a new set of credentials for a server they aren't authenticated in the IDE yet.

So, on the first run it results in the following picture:
![image](https://github.com/user-attachments/assets/3d581ef5-359e-42d0-896e-437b4bed159b)

Reply:
Huh, you are right. It looks better in the main branch while no account exists, but it is still not functional.
![image](https://github.com/user-attachments/assets/48e62f7f-5263-49ed-9867-ef6c1949bf83)

Alright then, we'll manage that separately later on.